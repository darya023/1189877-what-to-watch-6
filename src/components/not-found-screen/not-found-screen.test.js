import React from 'react';
import {render, screen} from '@testing-library/react';
import {Route, Router, Switch} from 'react-router';
import {createMemoryHistory} from 'history';
import NotFoundScreen from './not-found-screen';
import userEvent from '@testing-library/user-event';

const history = createMemoryHistory();
history.push(`/test`);
it(`NotFoundScreen should redirect to Mainscreen when user clicks on logo`, () => {
  render(
      <Router history={history}>
        <Switch>
          <Route exact path={`/test`}>
            <NotFoundScreen />
          </Route>
          <Route exact path={`/`}><div>Mock Main Screen</div></Route>
        </Switch>
      </Router>
  );

  userEvent.click(screen.getAllByRole(`link`, {name: /W T W/i})[0]);
  expect(screen.getByText(/Mock Main Screen/i)).toBeInTheDocument();
  history.push(`/test`);
  userEvent.click(screen.getAllByRole(`link`, {name: /W T W/i})[1]);
  expect(screen.getByText(/Mock Main Screen/i)).toBeInTheDocument();
});
