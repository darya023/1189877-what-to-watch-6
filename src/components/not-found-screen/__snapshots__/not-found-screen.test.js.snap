// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should NotFoundScreen render correctly 1`] = `
<div>
  <div
    class="user-page user-page--fill"
  >
    <header
      class="page-header user-page__head"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      
    </header>
    <main
      class="user-page__content"
    >
      <div
        class="page"
      >
        <div
          class="page__text"
        >
          <h1>
            404
          </h1>
          <p>
            Page not found
          </p>
        </div>
        <div
          class="page__btn"
        >
          <a
            class="page-btn"
            href="/"
          >
            Go to homepage
          </a>
        </div>
      </div>
    </main>
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link"
          href="/"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
